File: clients/csharp/examples/collections-find-one-and-replace-nested.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Replace a document
    var filterBuilder = Builders<Document>.CollectionFilter;
    var filter = filterBuilder.Eq("metadata.language", "English");
    var replacement = new Document()
    {
      { "title", "Hidden Shadows of the Past" },
      { "number_of_pages", 481 },
      {
        "genres",
        new[] { "Biography", "Graphic Novel", "Dystopian", "Drama" }
      },
      {
        "metadata",
        new Document
        {
          { "isbn", "978-1-905585-40-3" },
          { "language", "French" },
          { "edition", "Anniversary Edition" },
        }
      },
    };
    var result = await collection.FindOneAndReplaceAsync(
      filter,
      replacement
    );

    Console.WriteLine(JsonSerializer.Serialize(result));
  }
}

File: clients/java/examples/collections-find-one-and-replace-nested.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Filter;
import com.datastax.astra.client.core.query.Filters;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Replace a document
    Filter filter = Filters.eq("metadata.language", "English");
    Document newDocument =
        new Document()
            .append("title", "Hidden Shadows of the Past")
            .append("number_of_pages", 481)
            .append("genres", List.of("Biography", "Graphic Novel", "Dystopian", "Drama"))
            .append(
                "metadata",
                Map.of(
                    "isbn", "978-1-905585-40-3",
                    "language", "French",
                    "edition", "Anniversary Edition"));
    Optional<Document> result = collection.findOneAndReplace(filter, newDocument);
    System.out.println(result);
  }
}
